dbus.go

```diff
@@ -288,3 +332,74 @@ func (ss *SecretService) attrsMatch(attrs map[string]string, itemPath dbus.Objec
 	}
 	return true, nil
 }
+
+func (ss *SecretService) unlockItem(itemPath dbus.ObjectPath) (dbus.ObjectPath, error) {
+	unlocked, promptPath, err := ss.unlock([]dbus.ObjectPath{itemPath})
+	if err != nil {
+		return "", err
+	}
+
+	if len(unlocked) > 0 {
+		// we'll never get back more than 1 item in the slice
+		return unlocked[0], nil
+	}
+
+	if promptPath == dbus.ObjectPath("/") {
+		return "", fmt.Errorf("huh? no item or prompt when unlocking: %v", itemPath)
+	}
+
+	if ok, err := ss.prompt(promptPath); err != nil {
+		return "", err
+	} else if !ok {
+		return "", errors.New("unlock cancelled by user")
+	}
+	return itemPath, nil
+}
+
+// unlock attempts to unlock the objects given and returns the paths for the
+// objects that were unlocked and a prompt path to unlock the remainder.
```

Comment:
Add link to docs (below / above as well)

Reply:
Done for unlocking and prompting - hopefully this gives some background to help understand these operations.